addons/point_of_sale/static/src/app/models/pos_order.js

```diff
@@ -16,9 +16,17 @@ export class PosOrder extends Base {
 
     setup(vals) {
         super.setup(vals);
+        this.uiState = this.uiState ?? {};
```

Comment:
```suggestion
        let isResidual = false;
```

and in the following it, compute it's actual value.